File: server/router/internal_system/future_data/future_delivery_detail_by_input.go
package future_data

import (
	v1 "gin-vue-admin/api/v1/internal_system/future_data"
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

func InitFutureDeliveryDetailByInputRouter(Router *gin.RouterGroup) {
	FutureDeliveryDetailByInputRouter := Router.Group("futureDeliveryDetailByInput").Use(middleware.OperationRecord())
	{
		FutureDeliveryDetailByInputRouter.POST("createFutureDeliveryDetailByInput", v1.CreateFutureDeliveryDetailByInput)             // 新建FutureDeliveryDetailByInput
		FutureDeliveryDetailByInputRouter.DELETE("deleteFutureDeliveryDetailByInput", v1.DeleteFutureDeliveryDetailByInput)           // 删除FutureDeliveryDetailByInput
		FutureDeliveryDetailByInputRouter.DELETE("deleteFutureDeliveryDetailByInputByIds", v1.DeleteFutureDeliveryDetailByInputByIds) // 批量删除FutureDeliveryDetailByInput
		FutureDeliveryDetailByInputRouter.PUT("updateFutureDeliveryDetailByInput", v1.UpdateFutureDeliveryDetailByInput)              // 更新FutureDeliveryDetailByInput
		FutureDeliveryDetailByInputRouter.GET("findFutureDeliveryDetailByInput", v1.FindFutureDeliveryDetailByInput)                  // 根据ID获取FutureDeliveryDetailByInput
		FutureDeliveryDetailByInputRouter.GET("getFutureDeliveryDetailByInputList", v1.GetFutureDeliveryDetailByInputList)            // 获取FutureDeliveryDetailByInput列表
	}
}

File: web/src/api/internalSystem/futureData/futureDeliveryDetailByInput.js
import service from '@/utils/request'

// @Tags FutureDeliveryDetailByInput
// @Summary 创建FutureDeliveryDetailByInput
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.FutureDeliveryDetailByInput true "创建FutureDeliveryDetailByInput"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /futureDeliveryDetailByInput/createFutureDeliveryDetailByInput [post]
export const createFutureDeliveryDetailByInput = (data) => {
     return service({
         url: "/futureDeliveryDetailByInput/createFutureDeliveryDetailByInput",
         method: 'post',
         data
     })
 }


// @Tags FutureDeliveryDetailByInput
// @Summary 删除FutureDeliveryDetailByInput
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.FutureDeliveryDetailByInput true "删除FutureDeliveryDetailByInput"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /futureDeliveryDetailByInput/deleteFutureDeliveryDetailByInput [delete]
 export const deleteFutureDeliveryDetailByInput = (data) => {
     return service({
         url: "/futureDeliveryDetailByInput/deleteFutureDeliveryDetailByInput",
         method: 'delete',
         data
     })
 }

// @Tags FutureDeliveryDetailByInput
// @Summary 删除FutureDeliveryDetailByInput
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除FutureDeliveryDetailByInput"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /futureDeliveryDetailByInput/deleteFutureDeliveryDetailByInput [delete]
 export const deleteFutureDeliveryDetailByInputByIds = (data) => {
     return service({
         url: "/futureDeliveryDetailByInput/deleteFutureDeliveryDetailByInputByIds",
         method: 'delete',
         data
     })
 }

// @Tags FutureDeliveryDetailByInput
// @Summary 更新FutureDeliveryDetailByInput
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.FutureDeliveryDetailByInput true "更新FutureDeliveryDetailByInput"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /futureDeliveryDetailByInput/updateFutureDeliveryDetailByInput [put]
 export const updateFutureDeliveryDetailByInput = (data) => {
     return service({
         url: "/futureDeliveryDetailByInput/updateFutureDeliveryDetailByInput",
         method: 'put',
         data
     })
 }


// @Tags FutureDeliveryDetailByInput
// @Summary 用id查询FutureDeliveryDetailByInput
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.FutureDeliveryDetailByInput true "用id查询FutureDeliveryDetailByInput"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /futureDeliveryDetailByInput/findFutureDeliveryDetailByInput [get]
 export const findFutureDeliveryDetailByInput = (params) => {
     return service({
         url: "/futureDeliveryDetailByInput/findFutureDeliveryDetailByInput",
         method: 'get',
         params
     })
 }


// @Tags FutureDeliveryDetailByInput
// @Summary 分页获取FutureDeliveryDetailByInput列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取FutureDeliveryDetailByInput列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /futureDeliveryDetailByInput/getFutureDeliveryDetailByInputList [get]
 export const getFutureDeliveryDetailByInputList = (params) => {
     return service({
         url: "/futureDeliveryDetailByInput/getFutureDeliveryDetailByInputList",
         method: 'get',
         params
     })
 }